Rule builder

victorg1991 opened the pull request.

sandrodw3 commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

sandrodw3 commented:
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/63097. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

sandrodw3 commented:
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/63117. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

sandrodw3 commented:
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/63161. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

sandrodw3 commented:
Pull request submitted to https://github.com/p2kmgcl/liferay-portal/pull/174. [See changes here.](https://github.com/sandrodw3/liferay-portal/compare/5d7a4c6ccfa9~2...5d7a4c6ccfa9) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

pedro-oliveira446 committed:
LPS-194848 Add baseResourceURL parameter

pedro-oliveira446 committed:
LPS-194848 Add MVCResourceCommand to validate fields

pedro-oliveira446 committed:
LPS-194848 Add new language key

pedro-oliveira446 committed:
LPS-194848 Build Lang

pedro-oliveira446 committed:
LPS-194848 Show message with language

pedro-oliveira446 committed:
LPS-194848 Avoid null pointer

gislayne-vitorino committed:
LPS-198236 (dynamic-data-lists-test) Add test for DDMStorageLinkTableReferenceDefinition

igor-franca committed:
LPS-198010 Show Error Alert in ModalSelectObjectField

igor-franca committed:
LPS-198853 Use Clay Empty State in ModalSelectObjectField

igor-franca committed:
LPS-198853 Create the Necessary Language Keys to Modalselectedobjectfield Empty State and Improve an Existing One

igor-franca committed:
LPS-198853 buildLang

liaoethan committed:
ISSD-118 import primary nav adt for mega menu dropdown

liaoethan committed:
ISSD-118 add and refactor public sites nav to be used for CP

liaoethan committed:
ISSD-118 add navigation language menu adt

CalumR23 committed:
POSHI-630 Add new property to set browser resolution

CalumR23 committed:
POSHI-630 Add check to set window size

liaoethan committed:
ISSD-118 SF

liaoethan committed:
ISSD-118 update styles for CP nav

liaoethan committed:
ISSD-118 update CP master page to use public sites nav

ling-alan-huang committed:
LPS-196539 Rename

ling-alan-huang committed:
LPS-196539 Rename

igor-franca committed:
LPS-197264 Extract the Required Logic Out of the Component

igor-franca committed:
LPS-197264 Disable the Checkbox for a Persisted Object Field in a Composite Key Validation to Avoid Unchecking It in a Published Object Definition

BryanEngler committed:
LPS-199157 portal-search-tuning-rankings-web: We should reset the index even if the database is empty

https://liferay.atlassian.net/browse/LPS-199157

BryanEngler committed:
LPS-199157 portal-search-tuning-synonyms-web: We should reset the index even if the database is empty

https://liferay.atlassian.net/browse/LPS-199157

gislayne-vitorino committed:
LPS-198233 (dynamic-data-mapping-test) Add test for DDMFormInstanceReportTableReferenceDefinition

pedro-oliveira446 committed:
LPS-198010 Remove uppercase

pedro-oliveira446 committed:
LPS-198010 Rename

pedro-oliveira446 committed:
LPS-198010 Use lambda expression

pedro-oliveira446 committed:
LPS-198010 Inline

Commit:
ISSD-118 format CSS and adjust styles to use clay dropdown in Public Site Nav Fragment for CP

thektan committed:
LPS-191845 frontend-js-react-web: Fix react portal causing DOMException

The exception is caused when the container is unmounted before the children. This unmounts the react portal children before the page navigates.

thektan committed:
LPS-191845 frontend-data-set-web: Hide side panel from flashing open on initial load

liaoethan committed:
ISSD-118 add additional styles for CP

ISSD-118 style mega menu and refactor responsiveness

ISSD-118 additional changes to primary nav adt

ISSD-118 add animations to carets

pedro-oliveira446 committed:
LPS-198010 Reorder

pedro-oliveira446 committed:
LPS-198010 Rename

pedro-oliveira446 committed:
LPS-198010 Simplify

pedro-oliveira446 committed:
LPS-198010 Use the constants and inline

liaoethan committed:
ISSD-118 add navigation menu item custom fields to CP site initializer

liaoethan committed:
ISSD-118 change user default permissions to view custom fields

thektan committed:
LPS-191845 frontend-js-react-web: Add check if element exists before unmounting

liaoethan committed:
ISSD-118 SF

pei-jung committed:
LPS-200161 Support filtering CTEntries by workflow status

shuyangzhou committed:
LPS-199820 Defer Detector creation

pei-jung committed:
LPS-200161 Update CTEntry when its model is updated so a reindex can be triggered

BryanEngler committed:
LPS-199157 portal-search-tuning-rankings-web: Rename method

https://liferay.atlassian.net/browse/LPS-199157

BryanEngler committed:
LPS-199157 portal-search-tuning-rankings-web: Remove redundant check to see if index exists

https://liferay.atlassian.net/browse/LPS-199157

BryanEngler committed:
LPS-199157 portal-search-tuning-synonyms-web: Rename method

https://liferay.atlassian.net/browse/LPS-199157

BryanEngler committed:
LPS-199157 portal-search-tuning-synonyms-web: Remove redundant check to see if index exists

https://liferay.atlassian.net/browse/LPS-199157

liaoethan committed:
ISSD-118 accommodate nav for wider monitor screens

liaoethan committed:
ISSD-118 remove dropdown for nav menu item if there are no secondary items

igor-franca committed:
LPS-199932 Use Overflow to Hide horizontal scrollbar

peerkar committed:
LPS-199864 portal-search-api: Add Collapse API

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-api: Add export

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-api: Add Collapse to SearchRequest

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-api: gw baseLine

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search: Add Collapse Implementation classes

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search: Add Collapse to SearchRequest implementation

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-engine-adapter-api: Add Collapse to SearchSearchRequest

https://liferay.atlassian.net/browse/LPS-199864

david-truong committed:
LPS-198236 SF

peerkar committed:
LPS-199864 portal-search-engine-adapter-api: gw baseLine

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-elasticsearch7-impl: Add Collapse to the SearchSource

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-elasticsearch7-impl: Add Collapse to the SearchSearchRequest

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199864 portal-search-test: Add test case to the SearchRequestBuilder tests

https://liferay.atlassian.net/browse/LPS-199864

peerkar committed:
LPS-199873 search-experiences-rest-impl: Add Collapse to the advanced configuration schema

https://liferay.atlassian.net/browse/LPS-199873

peerkar committed:
LPS-199873: gw buildRest

https://liferay.atlassian.net/browse/LPS-199873

peerkar committed:
LPS-199873 search-experiences-service: Reorganize sort conversion for reusability

https://liferay.atlassian.net/browse/LPS-199873

peerkar committed:
LPS-199873 search-experiences-service: Handle Collapse

https://liferay.atlassian.net/browse/LPS-199873

peerkar committed:
LPS-199873 search-experiences-service: Add test

https://liferay.atlassian.net/browse/LPS-199873

peerkar committed:
LPS-199873 search-experiences-web: Add Collapse to Blueprint editor autocomplete schema

https://liferay.atlassian.net/browse/LPS-199873

Commit:
ISSD-118 bug fix for remove dropdown for nav menu item if there are no secondary items

antonio-ortega committed:
LPS-199436 This error is thrown by Safari and iOS when requesting an URL outside our domain. Since SPA can't redirect to outside Liferay in those cases, we are just forcing a manual redirect

liaoethan committed:
ISSD-118 submenu nav section spans full width

BryanEngler committed:
LPS-199864 portal-search: Sort

https://liferay.atlassian.net/browse/LPS-199864

liaoethan committed:
ISSD-118 SF

pei-jung committed:
LPS-200161 Index the status field with the correct name

pei-jung committed:
LPS-200161 Index the status label so it can be sorted

pei-jung committed:
LPS-200161 Allow changes to be filtered by workflow status in the Review Changes screen

thektan committed:
LPS-191845 frontend-js-react-web: Catch error so navigation doesn't break

david-truong committed:
LPS-198325 remove feature flag for LPS-180155 and deleting old publication views and leverage new FDS displays instead

ling-alan-huang committed:
LPS-196539 SF, rename

ling-alan-huang committed:
LPS-196539 SF, rename

ling-alan-huang committed:
LPS-196539 SF, rename

ling-alan-huang committed:
LPS-196539 SF, rename

david-truong committed:
LPS-198533 remove feature flag for LPS-147671

Commit:
LRAC-14553 Ignore tests affected by LRAC-14468

Commit:
LPS-194354 Removes general ignore from files and adds ignore in each test case

Commit:
LPS-200294 Remove the step to change display style

Commit:
LPS-200176 Remove unused reference field _teamLocalService in class UserModelDocumentContributor, its usages was removed by a0243a838170b03adfa7c993e0e2e715f2b0a517

Commit:
LPS-200176 Remove unused reference field _wikiNodeModelResourcePermission in class EditPageMVCRenderCommand, its usages was removed by 7169ebf36b42a32a6850169ac628c4ea3c068738

Commit:
LPS-200176 Remove unused reference field _accountEntryUserRelLocalService in class PunchOutSessionResourceImpl, its usages was removed by 0af8956911efc6807f2c9670c0cb8931c2a1fb11

Commit:
LPS-200176 Add module to SF properties file

Commit:
LPS-200294 Update locators for publications with dataset display

Commit:
LPS-200294 Update the macro to revert publication

Commit:
LPS-200294 Update macro to publish publication

Commit:
LPS-200294 Update macro to invite users

Commit:
LPS-200294 Refactor test IsLocalizedInPublicationsAdmin

Commit:
LPS-200294 Remove the unused test

Stephy6 committed:
LPS-200039 Remove AUI related code and use native js methods instead in user_groups.jsp in the users-admin-web module

Stephy6 committed:
LPS-200039 Remove AUI related code and use native js methods instead in roles.jsp in the users-admin-web module

Commit:
LPS-194354 Add ANY_DROPDOWN_LABEL path

Commit:
LPS-194354 Add IN_USE_BADGE path

Commit:
LPS-194354 Add DropdownMenuItem.viewDisabledItem macro

Commit:
LPS-194354 Add DataSetAdmin.selectFilterType macro

Commit:
LPS-194354 Move the old setup into the tests to add new tests with different setup

Commit:
LPS-194354 Add poshi tests related to Restructure filters selection story

Stephy6 committed:
LPS-200039 Remove AUI related code and use native js methods instead in organizations.jsp in the users-admin-web module

Stephy6 committed:
LPS-200039 Remove AUI related code and use native js methods instead in sites.jsp in the users-admin-web module

Commit:
LPS-194354 SF

Commit:
LPS-194354 Remove duplicate macro

pei-jung committed:
LPS-200161 Make status column sortable

pei-jung committed:
LPS-200161 Fix sorting by site name

pei-jung committed:
LPS-200161 Add tests for filter and sort CTEntries by status

pei-jung committed:
LPS-200161 Add test for sort CTEntries by site name

joshitagaki committed:
LRAC-14813 Add macro and path for page list metrics

joshitagaki committed:
LRAC-14813 Add macro and path for audience card total visitor number

joshitagaki committed:
LRAC-14813 Add test metric values consistent between cards

CalumR23 committed:
POSHI-661 Fix SF unit tests with new Poshi syntax

Commit:
LPS-200294 Refactor test to filter published publication

Commit:
LPS-200294 Refactor macro to unschedule Publication

Commit:
LPS-200294 Refactor macro to view invited users

Commit:
LPS-200294 Update step to edit publication

Commit:
LPS-200294 Update the locator and test for publication order

sandrodw3 committed:
LPS-197957 Fix typo

victorg1991 committed:
LPS-197957 Convert draft service fetch to typescript

sandrodw3 committed:
LPS-197957 Add State and improve LayoutData and Rule types

sandrodw3 committed:
LPS-197957 Rename to getUsersURL

sandrodw3 committed:
LPS-197957 Move actions and conditions states to RulesModal component

sandrodw3 committed:
LPS-197957 Add action command to delete a rule

Co-authored-by: Lourdes Fernández Besada <lourdes.fernandez@liferay.com>

sandrodw3 committed:
LPS-197957 Add action command to add a rule

Co-authored-by: Lourdes Fernández Besada <lourdes.fernandez@liferay.com>

sandrodw3 committed:
LPS-197957 Send URLs

sandrodw3 committed:
LPS-197957 Add action command to update a rule

Co-authored-by: Lourdes Fernández Besada <lourdes.fernandez@liferay.com>

sandrodw3 committed:
LPS-197957 Add thunks to add, update and delete a rule

sandrodw3 committed:
LPS-197957 Dispatch thunks instead of actions

sandrodw3 committed:
LPS-197959 Add undo/redo for rules

sandrodw3 committed:
LPS-197957 Rename to pageRules

sandrodw3 committed:
LPS-197959 Types

victorg1991 committed:
LPS-198741 Add Fragments selector

victorg1991 committed:
LPS-198741 Add Role selector

victorg1991 committed:
LPS-198741 Wrap in role group

victorg1991 committed:
LPS-198741 Allow passing aria label

victorg1991 committed:
LPS-198741 Add language keys

victorg1991 committed:
LPS-198741 buildLang

victorg1991 committed:
LPS-198741 Save conditionType

victorg1991 committed:
LPS-198741 Fix copy

victorg1991 committed:
LPS-198741 Rename

victorg1991 committed:
LPS-198741 Types

victorg1991 committed:
LPS-198741 Generate one action and one condition by default

magjed4289 committed:
LPS-200492 Refactor of ApplicationAPI.createAPIApplication()

Commit:
LPS-200494 Add parentesis to avoid a malformed query with OR and AND operators

victorg1991 committed:
LPS-198741 Delete should behave different when having only one item

- When the first input has value, reset it to the default state
- When the first input doesn't have any value, don't show the delete button

marianoalvarosaiz committed:
LPS-200501 Once the startup is finished ResourceActions added by the PortletTracker should be added to each company.

igor-franca committed:
LPS-198853 Improve language key

igor-franca committed:
LPS-198853 buildlang

joshitagaki committed:
LRAC-14813 SF

igor-franca committed:
LPS-198853 Use the updated language key

mbowerman committed:
LPS-200518 Add upgrade process to update the data in the
LayoutPageTemplateStructureRel table to account for the new package of
the ObjectEntrySingleFormVariationInfoCollectionProvider class

pedro-oliveira446 committed:
LPS-198010 Separate plural and singular language keys

pedro-oliveira446 committed:
LPS-198010 Build Lang

pedro-oliveira446 committed:
LPS-198010 Return the message according to the number of fields

liaoethan committed:
ISSD-118 add navigation menu configuration to CP site initializer

Commit:
LRAC-14842 Update the test to check if the interaction generated in DXP increased the number of individuals

Commit:
LRAC-14842 Remove the test from quarantine

ling-alan-huang committed:
COMMERCE-12633 Auto SF

ling-alan-huang committed:
COMMERCE-12676 Auto SF

ling-alan-huang committed:
LPS-177111 Generated

ling-alan-huang committed:
POSHI-661 Fix test cases

qz1010 committed:
LPS-195032 Wordsmith

qz1010 committed:
LPS-195032 buildLang

Commit:
LPS-200294 Refactor test CanSwitchInviteeRoles

Commit:
LPS-200294 Refactor macro to view out of date publication

Commit:
LPS-200294 Update macro to view admin role name after upgrading

joshitagaki committed:
LRAC-14813 Update macro name for clear usage

shuyangzhou committed:
LPS-200602 Drop per company based concurrent execution to avoid db level deadlocks. This is undoing LPS-135731, with the new per Portlet based concurrent execution, we should still be fine even with large number of companies.

susanchen3 committed:
LPS-200501 Create test to add virtual instance with config file when db partitioning is enabled

susanchen3 committed:
LPS-200501 Add test to relevant

brookedalton committed:
LPS-133975 Create test for SystemEventTableReferenceDefinition

pyoo47 committed:
LRCI-3854 Print post content into console log if HTTP response code 422 is detected

pyoo47 committed:
LRCI-3854 Don't retry when 422 is detected

pyoo47 committed:
LRCI-3854 Send a message to slack if an unknown exception causes forwarding to fail

pyoo47 committed:
LRCI-3854 Enhance the success message that is posted to Slack

kenjiheigel committed:
POSHI-630 Set and use as a default property

shuyangzhou committed:
LPS-200671 Lazy creating ObjectMapper

liaoethan committed:
ISSD-118 import navigation menu item for sites dropdown

qz1010 committed:
LPS-195226 Auto SF

qz1010 committed:
COMMERCE-11942 SF

qz1010 committed:
COMMERCE-11942 Auto SF

Commit:
LPS-200294 Refactor test for publication template

Commit:
LPS-200294 Update steps to view invited users

Commit:
LPS-200294 Update steps to view the publications order

Commit:
LPS-200294 Update steps to assert scheduled publication

brianchandotcom committed:
U101

brianchandotcom committed:
LRCI-3854 prep next

brianchandotcom committed:
LPS-200518 SF

brianchandotcom committed:
POSHI-630 Rename

brianchandotcom committed:
POSHI-661 prep next

brianchandotcom committed:
LPS-166761 SF

brianchandotcom committed:
LPS-198010 Wordsmith

brianchandotcom committed:
LPS-198010 SF

brianchandotcom committed:
LPS-198010 Regen

veroglez committed:
LPS-200651 Add an empty alt for decorative images

jkappler committed:
LPS-200139 We now have a mapper for duplicated ERC, so adapt the test

jkappler committed:
LPS-200139 Take new permission into account

jkappler committed:
LPS-200139 Sort like in XML

Commit:
LPS-200335 Modify macro to wait for sidebar panel open

Commit:
LPS-200335 Use macro

victorg1991 committed:
LPS-198741 Extract Screen Reader announcer

victorg1991 committed:
LPS-198741 Allow moving between items with the arrow keys

victorg1991 committed:
LPS-198741 Types

victorg1991 committed:
LPS-198741 Show delete button when any of the item is focused

kenjiheigel committed:
POSHI-665 Restrict legacy-portal to com.liferay.poshi.runner:1.0.471

p2kmgcl committed:
LPS-198741 Extract onDeleteCondition

p2kmgcl committed:
LPS-198741 Keep a map of conditions

p2kmgcl committed:
LPS-198741 Focus previous/next condition when it is deleted

gaborlovasdotcom committed:
LPS-196993 Expanding macros

gaborlovasdotcom committed:
LPS-196993 Adding new test cases

kenjiheigel committed:
LPS-200350 Fix variable name

victorg1991 committed:
LPS-198741 Apply to actions too

victorg1991 committed:
LPS-198741 Add Feedback messages

sandrodw3 commented on modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page_rules/components/RuleBuilderSection.tsx:
```diff
@@ -33,11 +34,14 @@ const TriggerLabel = React.forwardRef<HTMLButtonElement, any>(
 
 type RuleBuilderActionProps = {
 	actions: ActionType[];
+	fragmentEntryLinks: FragmentEntryLink[];
```
Esto sobra no?

victorg1991 replied:
True

sandrodw3 commented on modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page_rules/components/RuleBuilderSection.tsx:
```diff
@@ -7,7 +7,7 @@ import ClayButton from '@clayui/button';
 import ClayIcon from '@clayui/icon';
 import ClayPanel from '@clayui/panel';
-import React from 'react';
+import React, {Dispatch, SetStateAction} from 'react';
```
No sabia que estaba el tipo SetStateAction este 🥲

victorg1991 replied:
El que estaba antes estaba fetén, pero este cubre los dos casos, cuando le pasas una funcion o un calor directamente

sandrodw3 commented on modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/common/components/ScreenReaderAnnouncer.tsx:
```diff
@@ -0,0 +1,67 @@
+			}, 10000);
+		}, []);
+
+		useImperativeHandle(ref, () => ({sendMessage}));
```
Uuuh poniendo en practica el cursete 😱

victorg1991 replied:
😏

sandrodw3 commented on modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments_and_widgets/components/reorder_sets_modal/KeyboardDragAndDropContext.tsx:
```diff
@@ -75,31 +72,18 @@ export function KeyboardDragAndDropContextProvider({
-		setTimeout(() => {
-			setTextMap((previousTextMap) => {
-				const nextTextMap = {...previousTextMap};
+	const sendMessage = useCallback((message) => {
```
Hace falta crear esta funcion? Por que no pasar directamente `screenReaderAnnouncerRef.current?.sendMessage`?

sandrodw3 commented on modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/page_rules/components/RuleBuilderSection.tsx:
```diff
@@ -138,11 +138,16 @@ export function RuleBuilderConditionSection({
+			conditionRefMap.get(nextCondition.id)?.focus();
+
+			setConditions((previousConditions) => {
+				return previousConditions.filter(
```
Aqui entraria arrow function no? Me da tirrilla el return ☹️
